packages/view3d/src/core/PostProcessing.ts

```diff
@@ -0,0 +1,198 @@
+import { SSAOPass } from "three/examples/jsm/postprocessing/SSAOPass";
+import { RenderPass } from "three/examples/jsm/postprocessing/RenderPass";
+import { UnrealBloomPass } from "three/examples/jsm/postprocessing/UnrealBloomPass";
+import { PerspectiveCamera, Scene, Vector2, WebGLRenderer } from "three";
+import { BokehPass } from "three/examples/jsm/postprocessing/BokehPass";
+import { SSRPass } from "three/examples/jsm/postprocessing/SSRPass";
+import { Reflector } from "three/examples/jsm/objects/ReflectorForSSRPass";
+import { Mesh } from "three/src/Three";
+import { ShaderPass } from "three/examples/jsm/postprocessing/ShaderPass";
+import { FXAAShader } from "three/examples/jsm/shaders/FXAAShader";
+
+
+import View3D from "../View3D";
+import { getObjectOption } from "../utils";
+
+export interface SsaoOptions {
+  kernelRadius: number;
+  kernelSize: number;
+  maxDistance: number;
+  minDistance: number;
+}
+
+export interface BloomOptions {
+  strength: number;
+  threshold: number;
+  radius: number;
+}
+
+export interface DoFOptions {
+  focus: number;
+  aperture: number;
+  maxblur: number;
+}
+
+export interface SsrOptions {
+  groundReflector: Reflector | null;
+  selects: Mesh[] | null;
+}
+
+class PostProcessing {
+  private _view3D: View3D;
+  private _composer: EffectComposer;
+  private _scene: Scene;
+  private _camera: PerspectiveCamera;
+  private _renderer: WebGLRenderer;
+  private _isPostProcessing: boolean;
+
+  // effect component
+  private _bloomComponent:  UnrealBloomPass | undefined;
+
+  public get composer() {
+    return this._composer;
+  }
+
+  public get bloomComponent(){ return this._bloomComponent; }
+  /**
+   * whether using post-processing.
+   * @type {boolean}
+   * @readonly
+   */
+  public get isPostProcessing() {
+    return this._isPostProcessing;
+  }
+
+  public constructor(view3D: View3D) {
+    this._view3D = view3D;
+    this._isPostProcessing = this._checkIsPostProcessing();
+    this._renderer = this._view3D.renderer.threeRenderer;
+    this._composer = new EffectComposer(this._renderer);
+
+    const scene = this._scene = this._view3D.scene.root;
+    const camera = this._camera = this._view3D.camera.threeCamera;
+
+    if (!this._isPostProcessing) return;
+
+    view3D.once("beforeRender", () => {
+      const {width, height} = this._view3D.renderer.canvasSize;
+
+      this.composer.setSize(width * window.devicePixelRatio, height * window.devicePixelRatio);
+
+      const renderPass = new RenderPass(scene, camera);
+      this.composer.addPass(renderPass);
+
+      if (this._view3D.ssr) {
+        this._addSsr(getObjectOption(this._view3D.ssr));
+      }
+
+      if (this._view3D.ssao) {
+        this._addSSao(getObjectOption(this._view3D.ssao));
+      }
+
+      if (this._view3D.dof) {
+        this._addDoF(getObjectOption(this._view3D.dof));
+      }
+
+      if (this._view3D.bloom) {
+        this._bloomComponent = this._addBloom(getObjectOption(this._view3D.bloom));
+      }
```

Comment:
How about creating & calling `postProcessing.init()` from View3D in this case?

Reply:
Check it!